spacy_udpipe/language.py

```diff
@@ -135,8 +140,13 @@ def __call__(
             dtype="uint64"
         )
         doc.from_array(attrs=[LEMMA], array=lemma_array)
-        doc.is_tagged = bool(any(pos) and any(tags))
-        doc.is_parsed = bool(any(deps))
+        if SPACY_V3:
+            for i, j in enumerate(morphs):
+                if j != "_" and j != "":
+                    doc[i].set_morph(j)
+        else:
+            doc.is_tagged = bool(any(pos) and any(tags))
```

Comment:
I think `Doc.has_annotation` is automatically added when adding `POS` `TAG` `DEP` or `HEAD`.

Reply:
It is. It would be more verbose and would not raise a warning if the check is done using `Doc.has_annotation` in the `if` part of the `if-else` block :)